minkindr_gtsam/src/rotation-quaternion-gtsam.cc

```diff
@@ -75,18 +75,33 @@ EVector3 inverseRotate(const EQuaternion& C, const EVector3& p) {
   return EVector3(&inverse_rotate_point, C, p);
 }
 
+template <typename Scalar_ = double>
+inline bool isLessThanEpsilons4thRoot(Scalar_ x){
+  static const Scalar_ epsilon4thRoot = pow(std::numeric_limits<Scalar_>::epsilon(), 1.0/4.0);
+  return x < epsilon4thRoot;
+}
+
 Eigen::Vector3d rotationLogImplementation(const RotationQuaternion& C,
                                           OptionalJacobian<3, 3> JC) {
   Eigen::Vector3d aa = C.log();
   if(JC) {
     double phi = aa.norm();
-    if(phi < 1e-10) {
+    if(phi == 0) {
       JC->setIdentity();
     } else {
-      double cot = - std::sin(phi)/(std::cos(phi)-1);
-      double a1 = 1/(phi*phi) * (1.0 - 0.5*phi*cot);
+
+      double phiAbs = fabs(phi);
       Eigen::Matrix3d px = kindr::minimal::skewMatrix(aa);
-      (*JC) = Eigen::Matrix3d::Identity() - 0.5 * px + a1 * px * px;
+
+      double a;
+      if(!isLessThanEpsilons4thRoot(phiAbs)) {
```

Comment:
No, I don't think we will need the forth order term.

This plot is just the abs(a_cot - a_poly), and I think the polynomial version is quite close to the true value as phi approaches 0.